### PR TITLE
Callable param provider

### PR DESCRIPTION
**Pull request by dantleech**
- [x] Allow any callable as a param provider
- [x] Made CHANGELOG an `.md` file.

### CODE REVIEW

File: lib/Benchmark/Remote/template/parameter_set_extractor.template [hunk 1]
```diff
@@ -22,7 +22,19 @@ $parameterSets = array();
 ob_start();
 
 foreach ($paramProviders as $paramProvider) {
-    $parameterSets[] = $benchmark->$paramProvider();
+    if (is_callable($paramProvider)) {
+        $parameterSets[] = call_user_func($paramProvider);
+        continue;
+    }
+
+    if (method_exists($class, $paramProvider)) {
```
Comment (dantleech) [inline lib/Benchmark/Remote/template/parameter_set_extractor.template]: Should be the other way round - local method to take precedence over global function f.e.

Comment (rybakit) [reply lib/Benchmark/Remote/template/parameter_set_extractor.template]: This check will give false positives for non-public methods: https://3v4l.org/j5kOr.

File: lib/Benchmark/Remote/template/parameter_set_extractor.template [hunk 2]
```diff
@@ -22,7 +22,19 @@ $parameterSets = array();
 ob_start();
 
 foreach ($paramProviders as $paramProvider) {
-    $parameterSets[] = $benchmark->$paramProvider();
+    if (is_callable($paramProvider)) {
+        $parameterSets[] = call_user_func($paramProvider);
```
Comment (rybakit) [inline lib/Benchmark/Remote/template/parameter_set_extractor.template]: Can be simplified/optimized to `$parameterSets[] = $paramProvider();`

File: lib/Benchmark/Remote/template/parameter_set_extractor.template (hunk 1 above)
Comment (dantleech) [reply lib/Benchmark/Remote/template/parameter_set_extractor.template]: I guess we could normalize it to `$className . '::' . $paramProvider` https://3v4l.org/YnBCN

Comment (rybakit) [reply lib/Benchmark/Remote/template/parameter_set_extractor.template]: Yep, either `is_callable("$className::$paramProvider")` or `is_callable([$benchmark, $paramProvider])` will work.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Features
+
+- Assertions [docs](http://phpbench.readthedocs.io/en/latest/writing-benchmarks.html#assertions)
+- Added `--disable-php-ini` option
+- Report if opcode extension is enabled in env.
+- Show enabled PHP extensions in env.
+- PHP 7 only
+
+## Bugfixes
```
Comment (dantleech) [inline CHANGELOG.md]: should be `###`

File: benchmarks/Micro/HashingBench.php [hunk 4]
```diff
@@ -12,24 +12,27 @@
...
-    }
-
-    public function benchSha256()
+    /**
+     * @ParamProviders({"\PhpBench\Benchmarks\Micro\hash_algos"})
+     */
+    public function benchAlgos($params)
     {
-        return hash('sha256', 'hello world');
+        return md5($params[0]);
```
Comment (dantleech) [inline benchmarks/Micro/HashingBench.php]: md5?

**Comment by dantleech** [comment]
:boom:

**Pull request merged**